catalog/dags/data_refresh/dag_factory.py

```diff
@@ -75,27 +74,12 @@ def create_data_refresh_dag(data_refresh: DataRefresh, external_dag_ids: Sequenc
     )
 
     with dag:
-        # Count estimate SQL for determining the number of rows in the API table. While
-        # this query is not exact, it is close to accurate since the API media table
-        # does not receive many (if any) updates/insertions/deletions once the data
-        # refresh is complete. In testing this was shown to be accurate and fast
-        # (~0.000002% off and 1/14000th of the execution time). However, the reality is
-        # that this count is *not* exact, so we show it as an estimate in reports.
-        count_sql = f"""
-        SELECT c.reltuples::bigint AS estimate
-        FROM   pg_class c
-        JOIN   pg_namespace n ON n.oid = c.relnamespace
-        WHERE  c.relname = '{data_refresh.media_type}'
-        AND    n.nspname = 'public';
-        """
+        es_host = es.get_es_host(environment=PRODUCTION)
 
         # Get the current number of records in the target API table
-        before_record_count = PGExecuteQueryOperator(
-            task_id="get_before_record_count",
-            conn_id=OPENLEDGER_API_CONN_ID,
-            sql=count_sql,
-            handler=single_value,
-            return_last=True,
+        before_record_count = es.get_record_count_group_by_sources(
```

Comment:
By default, the function name is used as a task ID (and since we call this twice, Airflow helpfully adds `__1` to the end of the second call). We can make the task names a bit more descriptive in the UI by using the following:
```suggestion
        before_record_count = es.get_record_count_group_by_sources.override(
            task_id="get_before_record_count"
        )(
```